plugins/modules/vmware_resource_pool.py

```diff
@@ -274,35 +287,132 @@ def state_add_rp(self):
+        # check the difference between the existing config and the new config
+        rp_spec = self.generate_rp_config()
+        resource_pool_obj = self.select_resource_pool()
+
+        if self.mem_shares and self.mem_shares != resource_pool_obj.config.memoryAllocation.shares.level:
+            changed = True
+            rp_spec.memoryAllocation.shares.level = self.mem_shares
+
+        if self.mem_allocation_shares and self.mem_shares == 'custom':
+            if self.mem_allocation_shares != resource_pool_obj.config.memoryAllocation.shares.shares:
+                changed = True
+                rp_spec.memoryAllocation.shares.shares = self.mem_allocation_shares
+
+        if self.mem_limit and self.mem_limit != resource_pool_obj.config.memoryAllocation.limit:
+            changed = True
+            rp_spec.memoryAllocation.limit = self.mem_limit
+
+        if self.mem_reservation and self.mem_reservation != resource_pool_obj.config.memoryAllocation.reservation:
+            changed = True
+            rp_spec.memoryAllocation.reservation = self.mem_reservation
+
+        if self.mem_expandable_reservations != resource_pool_obj.config.memoryAllocation.expandableReservation:
+            changed = True
+            rp_spec.memoryAllocation.expandableReservation = self.mem_expandable_reservations
+
+        if self.cpu_shares and self.cpu_shares != resource_pool_obj.config.cpuAllocation.shares.level:
+            changed = True
+            rp_spec.cpuAllocation.shares.level = self.cpu_shares
+
+        if self.cpu_allocation_shares and self.cpu_shares == 'custom':
+            if self.cpu_allocation_shares != resource_pool_obj.config.cpuAllocation.shares.shares:
+                changed = True
+                rp_spec.cpuAllocation.shares.shares = self.cpu_allocation_shares
+
+        if self.cpu_limit and self.cpu_limit != resource_pool_obj.config.cpuAllocation.limit:
+            changed = True
+            rp_spec.cpuAllocation.limit = self.cpu_limit
+
+        if self.cpu_reservation and self.cpu_reservation != resource_pool_obj.config.cpuAllocation.reservation:
+            changed = True
+            rp_spec.cpuAllocation.reservation = self.cpu_reservation
+
+        if self.cpu_expandable_reservations != resource_pool_obj.config.cpuAllocation.expandableReservation:
+            changed = True
+            rp_spec.cpuAllocation.expandableReservation = self.cpu_expandable_reservations
+
+        if self.module.check_mode:
+            self.module.exit_json(changed=changed)
+
+        resource_pool_obj.UpdateConfig(self.resource_pool, rp_spec)
+
+        resource_pool_config = self.generate_rp_config_return_value(True)
+        self.module.exit_json(changed=changed, resource_pool_config=resource_pool_config)
+
+    def state_remove_rp(self):
+        changed = True
+        result = None
+        if self.module.check_mode:
+            self.module.exit_json(changed=changed)
+
+        resource_pool_obj = self.select_resource_pool()
```

Comment:
Add a check if we fail to find resource pool

Reply:
I modified the resource pool existence checking process as the following like.

https://github.com/ansible-collections/community.vmware/blob/1c6a92fb1d459f7aff355e8e1e7a0adfb3a282e0/plugins/modules/vmware_resource_pool.py#L410-L415